CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/Helpers/Order/OrderManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using CheckerUI.Models;
+using CheckerUI.ViewModels;
+using Microsoft.AspNetCore.SignalR.Client;
+using Xamarin.Forms;
+
+
+//This Class is responsible for scattering an order item to its correct cooker line,
+//every time a new order enters the server it updates (currently)
+//the todo list and from there in response we create
+//a new order model and update the list of the correct line in a new order
+
+namespace CheckerUI.Helpers.Order
+{
+    public class OrderManager : BaseViewModel
+    {
+        private static int m_TotalOrdersCounterID = 0;
+        public Dictionary<int, OrderItemView> m_Orders { get; set; } = new Dictionary<int, OrderItemView>();
+
+        private ObservableCollection<ToDo> m_ToDoList;
+        private ObservableCollection<OrderItemView> m_HotLineOrders { get; set; } = new ObservableCollection<OrderItemView>();
+        public OrderManager(ObservableCollection<OrderItemView> i_OrdersListOfLine)
+        {
+            CreateSignal();
+            m_ToDoList = new ObservableCollection<ToDo>();
+            m_HotLineOrders = i_OrdersListOfLine;
+            App.HubConn.On<String, DateTime>("ReceiveList", (desc, date) =>
+            {
+                ToDo todo = new ToDo() { createdDate = date, description = desc };
+                m_ToDoList.Add(todo);
+                addToOrdersList(todo);
+            });
```

Comment:
Here you define for the Hub Connection what to do when it is triggered with ReceiveList event, as appropriate.